src/metabase/cmd/copy.clj

```diff
@@ -164,11 +164,11 @@
     ;; Sample Database, the correct details are reset automatically on every
     ;; launch (see [[metabase.sample-data/update-sample-database-if-needed!]]), and we don't support connecting other H2
     ;; Databases in prod anyway, so this ultimately shouldn't cause anyone any problems.
-    (if *copy-h2-database-details*
-      identity
-      (map (fn [database]
-             (cond-> database
-               (= (:engine database) "h2") (assoc :details "{}")))))
+    (map (fn [database]
+           (cond-> database
+             (or (:is_attached_dwh database)
+                 (and (not *copy-h2-database-details*)
+                      (= (:engine database) "h2"))) (assoc :details "{}"))))
```

Comment:
Previously this would return an `O(1)` function when `*copy-h2-database-details*` was set and an `O(n)` function otherwise. Now it always returns an `O(n)` function.

That is fine, because (a) `*copy-h2-database-details*` was never set outside of tests so the `O(1)` case never happened in production and (b) `n` is commonly very small.